test/_util/helpers.js

```diff
@@ -4,6 +4,77 @@ import { clearLog, getLog } from './logCall';
 
 /** @jsx h */
 
+
+const VOID_ELEMENTS = /^(area|base|br|col|embed|hr|img|input|link|meta|param|source|track|wbr)$/;
+
+const el = document.createElement('div');
+const innerHtmlDef = Object.getOwnPropertyDescriptor(el.ownerDocument.defaultView.Element.prototype, 'innerHTML');
+function encodeEntities(str) {
+	el.textContent = str;
+	return innerHtmlDef.get.call(el);
+}
+
+/**
+ * Serialize a DOM tree.
+ * Uses deterministic sorting where necessary to ensure consistent tests.
+ * @param {Element|Node} node	The root node to serialize
+ * @returns {String} html
+ */
+function serializeDomTree(node) {
+	if (node.nodeType === 3) {
+		return encodeEntities(node.data);
+	}
+	else if (node.nodeType === 8) {
+		return '<!--' + encodeEntities(node.data) + '-->';
+	}
+	else if (node.nodeType === 1 || node.nodeType === 9) {
+		let str = '<' + node.localName;
+		const attrs = [];
+		for (let i=0; i<node.attributes.length; i++) {
+			attrs.push(node.attributes[i].name);
+		}
+		attrs.sort();
+		for (let i=0; i<attrs.length; i++) {
```

Comment:
Do you think we could merge these two loops, into one? Would it make sense?

```js
for (let i=0; i<node.attributes.length; i++) {
    const name = attrs[i];
    const value = node.getAttribute(name);
    if (value == null) continue;
    if (!value && name==='class') continue;
    attrs.push(node.attributes[i].name + '="' + encodeEntities(value) + '"');
}
// This will sort the _entire_ list _compiled_ already
attrs.sort();

// Join them with an empty space
str += attrs.join(" ");
```

Reply:
Pff, I was too late 😄

Reply:
Oh sorry, I was too quick with the merge I guess :S